cmd/query/app/static_handler_test.go

```diff
@@ -113,6 +114,57 @@ func TestNewStaticAssetsHandlerErrors(t *testing.T) {
 	}
 }
 
+// This test is potentially intermittent
+func TestHotReloadUIConfigTempFile(t *testing.T) {
+	tmpfile, err := ioutil.TempFile("", "ui-config-hotreload.*.json")
+	assert.NoError(t, err)
+
+	tmpFileName := tmpfile.Name()
+	defer os.Remove(tmpFileName)
+
+	content, err := ioutil.ReadFile("fixture/ui-config-hotreload.json")
+	assert.NoError(t, err)
+
+	err = ioutil.WriteFile(tmpFileName, content, 0644)
+	assert.NoError(t, err)
+
+	h, err := NewStaticAssetsHandler("fixture", StaticAssetsHandlerOptions{
+		UIConfigPath: tmpFileName,
+	})
+	assert.NoError(t, err)
+
+	c := string(h.indexHTML)
+	assert.Contains(t, c, "About Jaeger")
+
+	// wait for the watcher to start
+	h.watching.Wait()
```

Comment:
I don't think the  test needs to do that. It can start static handler in  httptest.Server, and poll it for v1 of the config, then rewrite the config to v2 and poll again for v2. This way it won't need access to any internals of the handler.

Reply:
Having a single go routine removed the need for this one. I'd still not depend on the http server, as it would broaden the scope of this unit test.